mozilla/fx-private-relay
# fix MPP-2288: add InboundContact.last_inbound_type

# New feature description
[The design for the "Callers and SMS Senders"](https://www.figma.com/file/g829jRfj82xF2x9dHKbpTk/Phone-Number-Mask?node-id=394%3A8827) "Latest Activity" column shows a different icon if the last activity was a call or a text.
<img width="425" alt="image" src="https://user-images.githubusercontent.com/71928/184733827-70f41ae2-04f4-4f6f-a009-2aaed92991be.png">


# Screenshot (if applicable)
This code adds a new `InboundContact.last_inbound_type` field that shows up on the `GET /api/v1/inboundcontact/[{id}]` endpoints:
<img width="446" alt="image" src="https://user-images.githubusercontent.com/71928/184734005-99c70bcc-2ebc-4bac-841d-b1a53f68833c.png">

# How to test

1. `pytest`
3. Make sure your profile is set to "Allow ⁨Firefox Relay⁩ to keep a log of your callers and text senders."
4. Send a text to your relay number
5. Check http://127.0.0.1:8000/api/v1/inboundcontact/
   * [ ] Check that the `last_inbound_type` field shows up and is accurate
6. Use `PATCH http://127.0.0.1:8000/api/v1/inboundcontact/{id}` to set `blocked: true`
7. Call your relay number
   * [ ] Check that your call is rejected (still ugly "Application Error" message for now)
8. Check http://127.0.0.1:8000/api/v1/inboundcontact/ again
   * [ ] Check that the `last_inbound_type` is still `text` (from step 3)

# Checklist
